connection.cpp

```diff
@@ -258,6 +261,83 @@ void UDPThread::run() {
   close(m_udpSocket);
 }
 
+
+bool
+UDPThread::handleMessage(uint8_t * buffer , size_t bufferSize) {
+
+  bool drop = false;
+  struct UDPDataPacket *data = NULL;
+  bool error = false;
+  /* Check for OP Code */
+  data = (struct UDPDataPacket*) buffer;
+  if (data->version != CANNELLONI_FRAME_VERSION) {
+    lwarn << "Received wrong version" << std::endl;
+    drop = true;
+  }
+  if (data->op_code != DATA) {
+    lwarn << "Received wrong OP code" << std::endl;
+    drop = true;
+  }
+  if (ntohs(data->count) == 0) {
+    linfo << "Received empty packet" << std::endl;
+    drop = true;
+  }
+  if (!drop) {
+    uint8_t *rawData = buffer+UDP_DATA_PACKET_BASE_SIZE;
+
+    if (m_debugOptions.udp) {
+      linfo << "Received " << std::dec << bufferSize << " Bytes from Host " << getAddressString (&m_remoteAddr)
+        << ":" << ntohs(m_remoteAddr.sin_port) << std::endl;
+    }
+    m_rxCount++;
+    for (uint16_t i=0; i<ntohs(data->count); i++) {
+      if (rawData-buffer+CANNELLONI_FRAME_BASE_SIZE > bufferSize) {
+        lerror << "Received incomplete packet" << std::endl;
+        error = true;
+      }
+      /* We got at least a complete canfd_frame header */
+      canfd_frame *frame = m_canThread->getFrameBuffer()->requestFrame();
+      if (!frame) {
+        lerror << "Allocation error." << std::endl;
+        error = true;
+      }
+      frame->can_id = ntohl(*((canid_t*)rawData));
+      /* += 4 */
+      rawData += sizeof(canid_t);
+      frame->len = *rawData;
+      /* += 1 */
+      rawData += sizeof(frame->len);
+      /* If this is a CAN FD frame, also retrieve the flags */
+      if (frame->len & CANFD_FRAME) {
+        frame->flags = *rawData;
+        /* += 1 */
+        rawData += sizeof(frame->flags);
+      }
+      /* RTR Frames have no data section although they have a dlc */
+      if ((frame->can_id & CAN_RTR_FLAG) == 0) {
+        /* Check again now that we know the dlc */
+        if (rawData-buffer+canfd_len(frame) > bufferSize) {
+          lerror << "Received incomplete packet / can header corrupt!" << std::endl;
+          error = true;
+        }
+        memcpy(frame->data, rawData, canfd_len(frame));
+        rawData += canfd_len(frame);
+      }
+      m_canThread->transmitCANFrame(frame);
+      if (m_debugOptions.can) {
+        printCANInfo(frame);
+      }
+    }
+  } else {
+    if(m_bind2firstConnection) {
```

Comment:
here should be `error |= drop;` since a dropped packet has been dropped due errors in the header.